net/src/main/java/io/bitcoinsv/jcl/net/protocol/handlers/message/MessageHandlerImpl.java

```diff
@@ -238,8 +245,23 @@ private void onStreamClosed(PeerAddress peerAddress) {
     // Event Handler:
     private void onStreamError(PeerAddress peerAddress, StreamErrorEvent event) {
         // We request a Disconnection from this Peer...
-        logger.trace(peerAddress, "Error detected in Stream, requesting disconnection... ");
-        super.eventBus.publish(new DisconnectPeerRequest(peerAddress));
+        logger.trace(peerAddress, "Error detected in Stream");
+        super.eventBus.publish(event); //re-publish the event to event bus (to allow further consumption & processing)
```

Comment:
have you checked a potential infinite-loop here? You are consuming an event and then publishing it again, so you might fall into an infinite consume/publish loop

Reply:
This re-publishing is now removed